In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt 
import seaborn as sns 

In [5]:
f_p='visa.csv'
visa_df=pd.read_csv(f_p)
visa_df

case_id continent education_of_employee has_job_experience  \
0         EZYV01      Asia           High School                  N   
1         EZYV02      Asia              Master's                  Y   
2         EZYV03      Asia            Bachelor's                  N   
3         EZYV04      Asia            Bachelor's                  N   
4         EZYV05    Africa              Master's                  Y   
...          ...       ...                   ...                ...   
25475  EZYV25476      Asia            Bachelor's                  Y   
25476  EZYV25477      Asia           High School                  Y   
25477  EZYV25478      Asia              Master's                  Y   
25478  EZYV25479      Asia              Master's                  Y   
25479  EZYV25480      Asia            Bachelor's                  Y   

      requires_job_training  no_of_employees  yr_of_estab  \
0                         N            14513         2007   
1                         N             2412         2002   
2                         Y            44444         2008   
3                         N               98         1897   
4                         N             1082         2005   
...                     ...              ...          ...   
25475                     Y             2601         2008   
25476                     N             3274         2006   
25477                     N             1121         1910   
25478                     Y             1918         1887   
25479                     N             3195         1960   

      region_of_employment  prevailing_wage unit_of_wage full_time_position  \
0                     West         592.2029         Hour                  Y   
1                Northeast       83425.6500         Year                  Y   
2                     West      122996.8600         Year                  Y   
3                     West       83434.0300         Year                  Y   
4                    South      149907.3900         Year                  Y   
...                    ...              ...          ...                ...   
25475                South       77092.5700         Year                  Y   
25476            Northeast      279174.7900         Year                  Y   
25477                South      146298.8500         Year                  N   
25478                 West       86154.7700         Year                  Y   
25479              Midwest       70876.9100         Year                  Y   

      case_status  
0          Denied  
1       Certified  
2          Denied  
3          Denied  
4       Certified  
...           ...  
25475   Certified  
25476   Certified  
25477   Certified  
25478   Certified  
25479   Certified  

[25480 rows x 12 columns]

In [6]:
visa_df.columns

Index(['case_id', 'continent', 'education_of_employee', 'has_job_experience',
       'requires_job_training', 'no_of_employees', 'yr_of_estab',
       'region_of_employment', 'prevailing_wage', 'unit_of_wage',
       'full_time_position', 'case_status'],
      dtype='object')

In [7]:
visa_df.select_dtypes(exclude='object').columns

Index(['no_of_employees', 'yr_of_estab', 'prevailing_wage'], dtype='object')

$PREVALLING WAGES$

- len
- count
- mean
- median
- min
- max
- std

In [8]:
visa_df['prevailing_wage']

0           592.2029
1         83425.6500
2        122996.8600
3         83434.0300
4        149907.3900
            ...     
25475     77092.5700
25476    279174.7900
25477    146298.8500
25478     86154.7700
25479     70876.9100
Name: prevailing_wage, Length: 25480, dtype: float64

In [10]:
len(visa_df['prevailing_wage'])

25480

In [11]:
visa_df['prevailing_wage'].count()

25480

$MEAN$

In [13]:
visa_df['prevailing_wage'].mean()  # pandas

74455.81459209183

In [12]:
np.mean(visa_df['prevailing_wage'])

74455.81459209183

$MEDIAN$

In [15]:
visa_df['prevailing_wage'].median()


70308.20999999999

In [16]:
np.median(visa_df['prevailing_wage'])

70308.20999999999

$min$

In [17]:
visa_df['prevailing_wage'].min()

2.1367

In [18]:
np.min(visa_df['prevailing_wage'])

2.1367

$max$

In [19]:
visa_df['prevailing_wage'].max()

319210.27

In [20]:
np.max(visa_df['prevailing_wage'])

319210.27

$standard deaviation$`

In [21]:
np.std(visa_df['prevailing_wage'])

52814.90589711402

In [22]:
visa_df['prevailing_wage'].std()

52815.94232687357

In [24]:
## all together
wage_count=round(visa_df['prevailing_wage'].count(),2)
wage_max=round(visa_df['prevailing_wage'].max(),2)
wage_min=round(visa_df['prevailing_wage'].min(),2)
wage_mean=round(visa_df['prevailing_wage'].mean(),2)
wage_median=round(visa_df['prevailing_wage'].median(),2)
wage_std=round(visa_df['prevailing_wage'].std(),2)

l=[wage_count,wage_max,wage_mean,wage_median,wage_min,wage_std]
cols=['prevailing_wage']
index=['count','max','mean','median','min','std']
pd.DataFrame(l,columns=cols,index=index)

prevailing_wage
count          25480.00
max           319210.27
mean           74455.81
median         70308.21
min                2.14
std            52815.94

$Percentile-quantile$

- percentile and quantile available in numpy

- np.percentile()

    - column name()
    - percentile values between 0 to 100
- np.quantile()
    - column name
    - 0 to 100
- in quantile 0.25 means 25 in percentile

In [29]:
np.percentile(visa_df['prevailing_wage'],25)

40.721391249999996

In [28]:
np.quantile(visa_df['prevailing_wage'],0.25)

34015.479999999996

**What is the mean of 25 percentile=34015**
- 25 percentage of total data has wage less than 34014.47
- find 25 percentage of total data=25*25480/100=6370
- 6370 person wage has lessthan 34014

In [32]:
con=visa_df['prevailing_wage']<34014
visa_df[con]

case_id      continent education_of_employee has_job_experience  \
0         EZYV01           Asia           High School                  N   
7         EZYV08  North America            Bachelor's                  Y   
12        EZYV13           Asia            Bachelor's                  Y   
16        EZYV17         Europe              Master's                  Y   
17        EZYV18           Asia              Master's                  Y   
...          ...            ...                   ...                ...   
25461  EZYV25462           Asia              Master's                  Y   
25465  EZYV25466  North America           High School                  N   
25466  EZYV25467         Europe            Bachelor's                  Y   
25470  EZYV25471  North America              Master's                  Y   
25473  EZYV25474           Asia            Bachelor's                  Y   

      requires_job_training  no_of_employees  yr_of_estab  \
0                         N            14513         2007   
7                         N             3035         1924   
12                        N           123876         1963   
16                        N            76638         1991   
17                        N             2747         2001   
...                     ...              ...          ...   
25461                     N             2861         2004   
25465                     N             2577         1995   
25466                     N             1938         2005   
25470                     N             2272         1970   
25473                     N             2421         2007   

      region_of_employment  prevailing_wage unit_of_wage full_time_position  \
0                     West         592.2029         Hour                  Y   
7                     West         418.2298         Hour                  Y   
12               Northeast       28663.0500         Year                  Y   
16                 Midwest        3706.7900         Year                  Y   
17                    West       16132.6100         Year                  Y   
...                    ...              ...          ...                ...   
25461                 West          54.9196         Hour                  Y   
25465                South         481.2238         Hour                  Y   
25466                 West        6973.4000         Year                  Y   
25470            Northeast         516.4101         Hour                  Y   
25473            Northeast       22845.5600         Year                  Y   

      case_status  
0          Denied  
7          Denied  
12      Certified  
16      Certified  
17      Certified  
...           ...  
25461      Denied  
25465   Certified  
25466      Denied  
25470   Certified  
25473   Certified  

[6370 rows x 12 columns]

In [33]:
len(visa_df[con])

6370

In [38]:
wage_50=np.percentile(visa_df['prevailing_wage'],50)


In [39]:
con=visa_df['prevailing_wage']<wage_50
visa_df[con]

case_id      continent education_of_employee has_job_experience  \
0         EZYV01           Asia           High School                  N   
6         EZYV07           Asia            Bachelor's                  N   
7         EZYV08  North America            Bachelor's                  Y   
9         EZYV10         Europe             Doctorate                  Y   
12        EZYV13           Asia            Bachelor's                  Y   
...          ...            ...                   ...                ...   
25465  EZYV25466  North America           High School                  N   
25466  EZYV25467         Europe            Bachelor's                  Y   
25470  EZYV25471  North America              Master's                  Y   
25473  EZYV25474           Asia            Bachelor's                  Y   
25474  EZYV25475         Africa             Doctorate                  N   

      requires_job_training  no_of_employees  yr_of_estab  \
0                         N            14513         2007   
6                         N             4985         1994   
7                         N             3035         1924   
9                         N             2251         1995   
12                        N           123876         1963   
...                     ...              ...          ...   
25465                     N             2577         1995   
25466                     N             1938         2005   
25470                     N             2272         1970   
25473                     N             2421         2007   
25474                     N             2594         1979   

      region_of_employment  prevailing_wage unit_of_wage full_time_position  \
0                     West         592.2029         Hour                  Y   
6                    South       53635.3900         Year                  Y   
7                     West         418.2298         Hour                  Y   
9                    South       67514.7600         Year                  Y   
12               Northeast       28663.0500         Year                  Y   
...                    ...              ...          ...                ...   
25465                South         481.2238         Hour                  Y   
25466                 West        6973.4000         Year                  Y   
25470            Northeast         516.4101         Hour                  Y   
25473            Northeast       22845.5600         Year                  Y   
25474            Northeast       51104.7800         Year                  Y   

      case_status  
0          Denied  
6       Certified  
7          Denied  
9       Certified  
12      Certified  
...           ...  
25465   Certified  
25466      Denied  
25470   Certified  
25473   Certified  
25474   Certified  

[12740 rows x 12 columns]

In [41]:
wage_count=round(visa_df['prevailing_wage'].count(),2)
wage_max=round(visa_df['prevailing_wage'].max(),2)
wage_min=round(visa_df['prevailing_wage'].min(),2)
wage_mean=round(visa_df['prevailing_wage'].mean(),2)
wage_median=round(visa_df['prevailing_wage'].median(),2)
wage_std=round(visa_df['prevailing_wage'].std(),2)
wage_25=np.percentile(visa_df['prevailing_wage'],25)
wage_50=np.percentile(visa_df['prevailing_wage'],50)
wage_75=np.percentile(visa_df['prevailing_wage'],75)


l=[wage_count,wage_max,wage_mean,wage_median,wage_min,wage_std,wage_25,wage_50,wage_75]
cols=['prevailing_wage']
index=['count','max','mean','median','min','std','25%','50%','75%']
pd.DataFrame(l,columns=cols,index=index)

prevailing_wage
count        25480.0000
max         319210.2700
mean         74455.8100
median       70308.2100
min              2.1400
std          52815.9400
25%          34015.4800
50%          70308.2100
75%         107735.5125

In [42]:
visa_df.describe()

no_of_employees   yr_of_estab  prevailing_wage
count     25480.000000  25480.000000     25480.000000
mean       5667.043210   1979.409929     74455.814592
std       22877.928848     42.366929     52815.942327
min         -26.000000   1800.000000         2.136700
25%        1022.000000   1976.000000     34015.480000
50%        2109.000000   1997.000000     70308.210000
75%        3504.000000   2005.000000    107735.512500
max      602069.000000   2016.000000    319210.270000

In [48]:
cols=visa_df.select_dtypes(exclude='object').columns
l=[]
for i in cols:
    
    count=round(visa_df[i].count(),2)
    maxx=round(visa_df[i].max(),2)
    minn=round(visa_df[i].min(),2)
    mean=round(visa_df[i].mean(),2)
    median=round(visa_df[i].median(),2)
    std=round(visa_df[i].std(),2)
    p_25=np.percentile(visa_df[i],25)
    p_50=np.percentile(visa_df[i],50)
    p_75=np.percentile(visa_df[i],75)


    l.append([count,maxx,mean,median,minn,std,p_25,p_50,p_75])
    
index=['count','max','mean','median','min','std','25%','50%','75%']
pd.DataFrame(zip(l[0],l[1],l[2]),columns=cols,index=index)

no_of_employees  yr_of_estab  prevailing_wage
count          25480.00     25480.00       25480.0000
max           602069.00      2016.00      319210.2700
mean            5667.04      1979.41       74455.8100
median          2109.00      1997.00       70308.2100
min              -26.00      1800.00           2.1400
std            22877.93        42.37       52815.9400
25%             1022.00      1976.00       34015.4800
50%             2109.00      1997.00       70308.2100
75%             3504.00      2005.00      107735.5125

In [50]:
cols=visa_df.select_dtypes(exclude='object').columns
d={}
for i in cols:
    
    count=round(visa_df[i].count(),2)
    maxx=round(visa_df[i].max(),2)
    minn=round(visa_df[i].min(),2)
    mean=round(visa_df[i].mean(),2)
    median=round(visa_df[i].median(),2)
    std=round(visa_df[i].std(),2)
    p_25=np.percentile(visa_df[i],25)
    p_50=np.percentile(visa_df[i],50)
    p_75=np.percentile(visa_df[i],75)


    d[i]=[count,maxx,mean,median,minn,std,p_25,p_50,p_75]
    
index=['count','max','mean','median','min','std','25%','50%','75%']
pd.DataFrame(d,columns=cols,index=index)

no_of_employees  yr_of_estab  prevailing_wage
count          25480.00     25480.00       25480.0000
max           602069.00      2016.00      319210.2700
mean            5667.04      1979.41       74455.8100
median          2109.00      1997.00       70308.2100
min              -26.00      1800.00           2.1400
std            22877.93        42.37       52815.9400
25%             1022.00      1976.00       34015.4800
50%             2109.00      1997.00       70308.2100
75%             3504.00      2005.00      107735.5125